cmd/collectors/volume.go

```diff
@@ -0,0 +1,208 @@
+			i.SetLabel(style, "flexgroup_constituent")
+			i.SetExportable(includeConstituents)
+		} else {
+			i.SetLabel(style, "flexvol")
+			key := i.GetLabel("svm") + "." + i.GetLabel("volume")
+			flexvolInstance, err := volumeAggrmetric.NewInstance(key)
+			if err != nil {
+				logger.Error().Err(err).Str("key", key).Msg("Failed to create new instance")
+				continue
+			}
+			flexvolInstance.SetLabels(maps.Clone(i.GetLabels()))
+			flexvolInstance.SetLabel(style, "flexvol")
+			if err := metric.SetValueFloat64(flexvolInstance, 1); err != nil {
+				logger.Error().Err(err).Str("metric", metricName).Msg("Unable to set value on metric")
+			}
+		}
+	}
+
+	logger.Debug().Int("flexgroup volume count", len(cache.GetInstances())).Send()
+
+	recordFGFalse := make(map[string]*set.Set)
+	for _, i := range data.GetInstances() {
+		match := flexgroupRegex.FindStringSubmatch(i.GetLabel("volume"))
+		if len(match) != 3 {
+			continue
+		}
+		key := i.GetLabel("svm") + "." + match[1]
+
+		flexgroupInstance := volumeAggrmetric.GetInstance(key)
+		if flexgroupInstance != nil {
+			aggrs := flexgroupAggrsMap[key].Values()
+			sort.Strings(aggrs)
+			flexgroupInstance.SetLabel("aggr", strings.Join(aggrs, ","))
+		}
+
+		fg := cache.GetInstance(key)
+		if fg == nil {
+			logger.Error().Msgf("instance [%s] not in local cache", key)
+			continue
+		}
+
+		aggrs := fgAggrMap[key].Values()
+		sort.Strings(aggrs)
+		fg.SetLabel("aggr", strings.Join(aggrs, ","))
+
+		for mkey, m := range data.GetMetrics() {
+			if !m.IsExportable() && m.GetType() != "float64" {
+				continue
+			}
+
+			fgm := cache.GetMetric(mkey)
+			if fgm == nil {
+				logger.Error().Msgf("metric [%s] not in local cache", mkey)
+				continue
+			}
+
+			if value, ok := m.GetValueFloat64(i); ok {
+				fgv, _ := fgm.GetValueFloat64(fg)
+
+				if !strings.HasSuffix(m.GetName(), "_latency") {
+					err := fgm.SetValueFloat64(fg, fgv+value)
+					if err != nil {
+						logger.Error().Err(err).Msg("error")
+					}
+					continue
+				}
+
+				opsKey := ""
+				if strings.Contains(mkey, "_latency") {
+					opsKey = m.GetComment()
+				}
+
+				if ops := data.GetMetric(opsKey); ops != nil {
+					if opsValue, ok := ops.GetValueFloat64(i); ok {
+						var tempOpsV float64
+
+						prod := value * opsValue
+						tempOpsKey := opsKeyPrefix + opsKey
+						tempOps := cache.GetMetric(tempOpsKey)
+
+						if tempOps == nil {
+							if tempOps, err = cache.NewMetricFloat64(tempOpsKey); err != nil {
+								return nil, nil, err
+							}
+							tempOps.SetExportable(false)
+						} else {
+							tempOpsV, _ = tempOps.GetValueFloat64(fg)
+						}
+						if value != 0 {
+							err = tempOps.SetValueFloat64(fg, tempOpsV+opsValue)
+							if err != nil {
+								logger.Error().Err(err).Msg("error")
+							}
+						}
+						err = fgm.SetValueFloat64(fg, fgv+prod)
+						if err != nil {
+							logger.Error().Err(err).Msg("error")
+						}
+					} else {
+						if recordFGFalse[key] == nil {
```

Comment:
this is slightly more idiomatic and does one lookup instead of two (after creation)
```go
s, ok := recordFGFalse[key]
if !ok {
	s = set.New()
	recordFGFalse[key] = s
}
s.Add(fgm.GetName())
```